e2e/playwright/viewer/src/tests/viewer-protected.spec.ts

```diff
@@ -0,0 +1,91 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { expect } from '@playwright/test';
+import { ApiClientFactory, getUserState, test, TEST_FILES, Utils } from '@alfresco/playwright-shared';
+
+test.use({ storageState: getUserState('hruser') });
+test.describe('viewer file', () => {
+  const apiClientFactory = new ApiClientFactory();
+  const randomFolderName = `playwright-folder-${Utils.random()}`;
+  const randomDocxName = TEST_FILES.DOCX.name + `-${Utils.random()}`;
+  let folderId: string;
+  let fileDocxId: string;
+
+  test.beforeAll(async ({ fileAction, shareAction, favoritesPageAction: favoritesPageAction }) => {
+    await apiClientFactory.setUpAcaBackend('hruser');
+    const node = await apiClientFactory.nodes.createNode('-my-', { name: randomFolderName, nodeType: 'cm:folder', relativePath: '/' });
+    folderId = await node.entry.id;
+    const fileDoc = await fileAction.uploadFile(TEST_FILES.DOCX_PROTECTED.path, randomDocxName, folderId);
+    fileDocxId = await fileDoc.entry.id;
+    await shareAction.shareFileById(fileDocxId);
+    await favoritesPageAction.addFavoriteById('file', fileDocxId);
+  });
+
+  test.beforeEach(async ({ personalFiles }) => {
+    await personalFiles.navigate({ waitUntil: 'domcontentloaded' });
+    await personalFiles.dataTable.performClickFolderOrFileToOpen(randomFolderName);
+    await personalFiles.dataTable.performClickFolderOrFileToOpen(randomDocxName);
+  });
+
+  test.afterAll(async () => {
+    await apiClientFactory.nodes.deleteNode(folderId);
```

Comment:
should we delete uploaded file, from beforeAll() at the end of testing ?

Reply:
the file will be deleted with folder, as its created inside the parent folder.